scipy/stats/_hypotests.py

```diff
@@ -1393,3 +1396,674 @@ def cramervonmises_2samp(x, y, method='auto'):
+        are considered to be paired; the third is for independent sample
+        statistics.
+
+        - ``'samples'`` : observations are assigned to different samples
+          but remain paired with the same observations from other samples.
+          This permutation type is appropriate for paired sample hypothesis
+          tests such as the Wilcoxon signed-rank test and the paired t-test.
+        - ``'pairings'`` : observations are paired with different observations,
+          but they remain within the same sample. This permutation type is
+          appropriate for association/correlation tests with statistics such
+          as Spearman's :math:`\rho`, Kendall's :math:`\tau`, and Pearson's
+          :math:`r`.
+        - ``'both'`` : observations are assigned to different samples without
+          preserving pairs. Samples may contain different numbers of
+          observations. This permutation type is appropriate for
+          independent sample hypothesis tests such as the Mann-Whitney
+          :math:`U` test and the independent sample t-test.
+
+          Please see the Notes section below for more detailed descriptions
+          of the permutation types.
+
+    vectorized : bool, default: ``False``
+        By default, `statistic` is assumed to calculate the statistic only for
+        1D arrays contained in `data`. If `vectorized` is set ``True``,
+        `statistic` must also accept a keyword argument `axis` and be
+        vectorized to compute the statistic along the provided `axis` of the ND
+        arrays in `data`. Use of a vectorized statistic can reduce computation
+        time.
+    permutations : int, default: ``np.inf``
+        Number of random permutations used to approximate the null
+        distribution. If greater than or equal to the number of distinct
+        permutations, the exact null distribution will be computed.
+        Note that the number of distinct permutations grows very rapidly with
+        the sizes of samples, so exact tests are feasible only for very small
+        data sets.
+    batch : int, optional
+        The number of permutations to process in each call to `statistic`.
+        Memory usage is O(`batch`*``n``), where ``n`` is the total size
+        of all samples, regardless of the value of `vectorized`. Default is
+        ``None``, in which case ``batch`` is the number of permutations.
+    alternative : {'two-sided', 'less', 'greater'}
+        The alternative hypothesis for which the p-value is calculated.
+        For each alternative, the p-value is defined as follows.
+
+        - ``'greater'`` : the percentage of the null distribution that is
+          greater than or equal to the observed value of the test statistic.
+        - ``'less'`` : the percentage of the null distribution that is
+          less than or equal to the observed value of the test statistic.
+        - ``'two-sided'`` : twice the smaller of the p-values above.
```

Comment:
This is the default value.

Reply:
This is implicit according to [numpydoc guidelines](https://numpydoc.readthedocs.io/en/latest/format.html#parameters):

> When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first

and above:

```
{‘two-sided’, ‘less’, ‘greater’}
```

But I see how it would be better to also make it explicit here, since it's in a different order. I plan to keep "two-sided" last _here_ (but first in the set above) because I think the one-sided p-values above should be described so it is easy to describe this one.